ada/testsuite/tests/name_resolution/requeue_stmt/test_requeue.adb

```diff
@@ -0,0 +1,93 @@
+with Ada.Text_IO        ; use Ada.Text_IO;
+with Ada.Integer_Text_IO; use Ada.Integer_Text_IO;
+
+procedure ProducerConsumer_Test is
+
+   Size : constant Positive := 5;
+
+   protected type Producer_Consumer is
+
+      entry     Get_Data (Value_Get : out Natural);
+      procedure Put_Data (Value_Put : in  Natural);
+
+   private
+      entry Wait_For_Get (Value_Get : out Natural);
+
+      Data_Available : Boolean := False;
+      The_Data       : Natural;
+
+   end Producer_Consumer;
+
+   protected body Producer_Consumer is
+
+      entry Get_Data (Value_Get : out Natural) when true is
+      begin
+         if Data_Available then
+            Value_Get := The_Data;
+            Data_Available := False;
+         else
+            requeue Wait_For_Get;
```

Comment:
Looking at the nameres logic you added, I think this is missing a few test cases!